### PR TITLE
set_bank takes owned Arc<Bank>

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
`set_bank` is taking ownership of the `Arc<Bank>`.

#### Summary of Changes
Update function signature.

Fixes #

### CODE REVIEW

File: core/src/replay_stage.rs [hunk 1]
```diff
@@ -1948,7 +1948,7 @@ impl ReplayStage {
             poh_recorder
                 .write()
                 .unwrap()
-                .set_bank(&tpu_bank, track_transaction_indexes);
+                .set_bank(tpu_bank, track_transaction_indexes);
```
Comment (ryoqun) [inline core/src/replay_stage.rs:1951]: yay, one less clone for block generation code path. :)

File: poh/src/poh_recorder.rs [hunk 2]
```diff
@@ -573,14 +573,14 @@ impl PohRecorder {
         self.leader_last_tick_height = leader_last_tick_height;
     }
 
-    pub fn set_bank(&mut self, bank: &Arc<Bank>, track_transaction_indexes: bool) {
+    pub fn set_bank(&mut self, bank: Arc<Bank>, track_transaction_indexes: bool) {
         assert!(self.working_bank.is_none());
-        self.leader_bank_notifier.set_in_progress(bank);
+        self.leader_bank_notifier.set_in_progress(&bank);
```
Comment (ryoqun) [inline poh/src/poh_recorder.rs:578]: hehe, `set_in_progress` seems to be a rare exception for this `Arc<_>` arg fixup saga... lol

**Review by ryoqun — APPROVED** [review]
lgtm; thanks for cleaning up (no nits this time!)

File: poh/src/poh_recorder.rs (hunk 2 above)
Comment (apfitzge) [reply poh/src/poh_recorder.rs:578]: Yeah, I think the exception is justified as it does not take ownership of the Arc. It uses the arc to create a weak-reference.

(I know you know this, just making it clear for anyone else looking at the PR why this **should** be an exception)

**Pull request merged**